src/main/java/com/artipie/rpm/misc/StorageLock.java

```diff
@@ -0,0 +1,144 @@
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.rpm.misc;
+
+import com.artipie.asto.Content;
+import com.artipie.asto.Key;
+import com.artipie.asto.Storage;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Locker for the storage.
+ * @since 0.9
+ * @todo #230:30min Use this class in RPM update to prohibit parallel update of the same repository:
+ *  add lock before starting the update and release it on terminate.
+ * @todo #230:30min Consider waiting for the lock to be released instead of throwing an exception.
+ *  This feature implementation should be properly discussed and planed first.
+ */
+public final class StorageLock {
+
+    /**
+     * Lock file name.
+     */
+    private static final String FNAME = "lock-%s";
+
+    /**
+     * Lock file name pattern.
+     */
+    private static final String PTRN = "%s/lock-[a-z0-9-]{36}";
+
+    /**
+     * Storage.
+     */
+    private final Storage storage;
+
+    /**
+     * Repository key.
+     */
+    private final Key repo;
+
+    /**
+     * Lock file key.
+     */
+    private final Key file;
+
+    /**
+     * Ctor.
+     * @param storage Storage
+     * @param repo Repo key
+     */
+    public StorageLock(final Storage storage, final Key repo) {
+        this.storage = storage;
+        this.repo = repo;
+        this.file = new Key.From(
+            this.repo, String.format(StorageLock.FNAME, UUID.randomUUID().toString())
+        );
+    }
+
+    /**
+     * Locks storage by adding random file to it.
+     * @return CompletableFuture
+     * @throws IllegalStateException If storage is already locked
+     */
+    public CompletableFuture<Void> lock() {
+        return this.countLocks().thenCompose(
+            count -> {
+                if (count == 0) {
+                    return this.storage.save(this.file, new Content.From(new byte[]{}))
+                        .<Void>thenCompose(
+                            ignored -> this.countLocks().<Void>thenCompose(
+                                cnt -> {
+                                    final CompletableFuture<Void> res;
+                                    if (cnt > 1) {
+                                        res = this.release().<Void>thenApply(
+                                            nothing -> {
+                                                throw this.error();
+                                            }
+                                        );
+                                    } else {
+                                        res = CompletableFuture.completedFuture(null);
+                                    }
+                                    return res;
+                                }
+                            )
+                        );
+                } else {
+                    throw this.error();
```

Comment:
@olenagerasimova can we remove this `error` method and just throw exceptions here? I think it make things less readable, I don't really see the need for introducing a method. Also you could add to the exception message the number of existing locks, to help debug in case of problem